app/views/student/comments/_form.html.haml

```diff
@@ -1,5 +1,6 @@
-= simple_form_for([:student, @course, post, post.comments.build]) do |f|
+= simple_form_for [:student,@course,post,post.comments.build] do |f|
```

Comment:
`post.comments.build` , I prefer  post.comments.new 
and should be in controller not in view